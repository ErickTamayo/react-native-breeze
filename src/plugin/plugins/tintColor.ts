import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = { key: string };

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^tint-(?<key>${keys("tintColor")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, color }) => {
  const { key } = groups;
  const value = color("tintColor", key);
  return { tintColor: value };
};
