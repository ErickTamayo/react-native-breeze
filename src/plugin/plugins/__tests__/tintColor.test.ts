import {
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../utils/tests";
import { pattern, plugin } from "../tintColor";

describe("tintColor", () => {
  const shouldMatch = ["tint-green-500", "tint-black"];

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["tint-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
