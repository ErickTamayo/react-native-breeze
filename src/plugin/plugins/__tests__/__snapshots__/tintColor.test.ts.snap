// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tintColor Should parse (tint-black) correctly 1`] = `
Object {
  "tintColor": "#000000",
}
`;

exports[`tintColor Should parse (tint-green-500) correctly 1`] = `
Object {
  "tintColor": "#48bb78",
}
`;
